src/components/NewCommentForm.tsx

```diff
@@ -1,37 +1,192 @@
-import React from 'react';
+import React, { useState } from 'react';
+import cn from 'classnames';
+import { CommentData } from '../types/Comment';
+import { ErrorMessage } from './ErrorMessage';
+
+interface ErrorMessages {
+  nameError: string,
+  emailError: string,
+  bodyError: string,
+  serverError: string,
+}
+
+const defaultErrorValues = {
+  nameError: '',
+  emailError: '',
+  bodyError: '',
+  serverError: '',
+};
+
+interface Props {
+  handleAddNewComment: (commentData: CommentData) => Promise<void>,
+}
+
+export const NewCommentForm: React.FC<Props> = ({
+  handleAddNewComment,
+}) => {
+  const [errors, setErrors] = useState<ErrorMessages>(defaultErrorValues);
+
+  const [commentData, setCommentData] = useState<CommentData>({
+    name: '',
+    email: '',
+    body: '',
+  });
+
+  const [isLoading, setIsLoading] = useState(false);
+
+  const handleReset = () => {
+    setCommentData({
+      name: '',
+      email: '',
+      body: '',
+    });
+    setErrors({
+      ...errors,
+      nameError: '',
+      emailError: '',
+      bodyError: '',
+    });
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const normalizedName = commentData.name.trim();
+    const normalizedEmail = commentData.email.trim();
+    const normalizedBody = commentData.body.trim();
+
+    let nameError = '';
+    let emailError = '';
+    let bodyError = '';
+
+    if (!normalizedName) {
+      nameError = 'Name is required';
+    }
+
+    if (!normalizedEmail) {
+      emailError = 'Email is required';
+    }
+
+    if (!normalizedBody) {
+      bodyError = 'Enter some text';
+    }
```

Comment:
Please try to rewrite for more readability. Try don't use let in this case. ( You can add all error case in enum). 
end add error text in the handlers.